src/Target.cpp

```diff
@@ -753,6 +774,103 @@ int Target::natural_vector_size(const Halide::Type &t) const {
     }
 }
 
+bool Target::get_runtime_compatible_target(const Target& other, Target &result) {
+    // A compatible runtime must support these features if _either_ of the base runtimes
+    static const std::vector<Feature> unionFeatures = {
+            // These features assist in debugging.
+            Debug, TSAN, ASAN, MSAN,
+
+            // These features enable access to other devices.
+            CUDA, CUDACapability30, CUDACapability32, CUDACapability35, CUDACapability50, CUDACapability61, OpenCL,
+            OpenGL, OpenGLCompute, MinGW, Metal, HexagonDma, HVX_64, HVX_128, HVX_v62, HVX_v65, HVX_v66,
+            HVX_shared_object, D3D12Compute
+    };
+
+    static const std::vector<Feature> intersectionFeatures = {
```

Comment:
Most of the features in the following two lists do not meaningfully affect the runtime and should just be stripped. Exceptions:

SoftFloatABI must match across all targets in the list, (like arch, os, bits)
NoNEON is a union feature